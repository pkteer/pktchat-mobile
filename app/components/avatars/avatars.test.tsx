// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import Preferences from '@mm-redux/constants/preferences';

import Avatars from './avatars';

describe('Avatars', () => {
    test('should match snapshot for single avatar', () => {
        const baseProps = {
            theme: Preferences.THEMES.denim,
            userIds: ['user1'],
        };

        const wrapper = shallow(<Avatars {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should match snapshot for overflow', () => {
        const baseProps = {
            theme: Preferences.THEMES.denim,
            userIds: ['user1', 'user2', 'user3', 'user4', 'user5', 'user6'],
        };

        const wrapper = shallow(<Avatars {...baseProps}/>);
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
