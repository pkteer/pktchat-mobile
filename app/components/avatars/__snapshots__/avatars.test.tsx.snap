// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatars should match snapshot for overflow 1`] = `
<TouchableWithFeedbackIOS
  onPress={[Function]}
  type="opacity"
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 12,
          "borderWidth": 13,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <Connect(ProfilePicture)
        showStatus={false}
        size={24}
        testID="avatars.profile_picture"
        theme={
          Object {
            "awayIndicator": "#ffbc1f",
            "buttonBg": "#1c58d9",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3f4350",
            "codeTheme": "github",
            "dndIndicator": "#d24b4e",
            "errorTextColor": "#d24b4e",
            "linkColor": "#386fe5",
            "mentionBg": "#ffffff",
            "mentionColor": "#1e325c",
            "mentionHighlightBg": "#ffd470",
            "mentionHighlightLink": "#1b1d22",
            "newMessageSeparator": "#cc8f00",
            "onlineIndicator": "#3db887",
            "sidebarBg": "#1e325c",
            "sidebarHeaderBg": "#192a4d",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarTeamBarBg": "#14213e",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#5d89ea",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#28427b",
            "sidebarUnreadText": "#ffffff",
            "type": "Denim",
          }
        }
        userId="user1"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 12,
          "borderWidth": 13,
          "height": 24,
          "justifyContent": "center",
          "marginLeft": -6,
          "width": 24,
        }
      }
    >
      <Connect(ProfilePicture)
        showStatus={false}
        size={24}
        testID="avatars.profile_picture"
        theme={
          Object {
            "awayIndicator": "#ffbc1f",
            "buttonBg": "#1c58d9",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3f4350",
            "codeTheme": "github",
            "dndIndicator": "#d24b4e",
            "errorTextColor": "#d24b4e",
            "linkColor": "#386fe5",
            "mentionBg": "#ffffff",
            "mentionColor": "#1e325c",
            "mentionHighlightBg": "#ffd470",
            "mentionHighlightLink": "#1b1d22",
            "newMessageSeparator": "#cc8f00",
            "onlineIndicator": "#3db887",
            "sidebarBg": "#1e325c",
            "sidebarHeaderBg": "#192a4d",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarTeamBarBg": "#14213e",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#5d89ea",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#28427b",
            "sidebarUnreadText": "#ffffff",
            "type": "Denim",
          }
        }
        userId="user2"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 12,
          "borderWidth": 13,
          "height": 24,
          "justifyContent": "center",
          "marginLeft": -6,
          "width": 24,
        }
      }
    >
      <Connect(ProfilePicture)
        showStatus={false}
        size={24}
        testID="avatars.profile_picture"
        theme={
          Object {
            "awayIndicator": "#ffbc1f",
            "buttonBg": "#1c58d9",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3f4350",
            "codeTheme": "github",
            "dndIndicator": "#d24b4e",
            "errorTextColor": "#d24b4e",
            "linkColor": "#386fe5",
            "mentionBg": "#ffffff",
            "mentionColor": "#1e325c",
            "mentionHighlightBg": "#ffd470",
            "mentionHighlightLink": "#1b1d22",
            "newMessageSeparator": "#cc8f00",
            "onlineIndicator": "#3db887",
            "sidebarBg": "#1e325c",
            "sidebarHeaderBg": "#192a4d",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarTeamBarBg": "#14213e",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#5d89ea",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#28427b",
            "sidebarUnreadText": "#ffffff",
            "type": "Denim",
          }
        }
        userId="user3"
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 13.5,
          "borderWidth": 1,
          "height": 27,
          "justifyContent": "center",
          "marginLeft": -6,
          "width": 27,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(63,67,80,0.08)",
            "borderColor": "#ffffff",
            "borderRadius": 13.5,
            "borderWidth": 1,
            "height": 27,
            "justifyContent": "center",
            "width": 27,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgba(63,67,80,0.64)",
              "fontSize": 10,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          +3
        </Text>
      </View>
    </View>
  </View>
</TouchableWithFeedbackIOS>
`;

exports[`Avatars should match snapshot for single avatar 1`] = `
<TouchableWithFeedbackIOS
  onPress={[Function]}
  type="opacity"
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 12,
          "borderWidth": 13,
          "height": 24,
          "justifyContent": "center",
          "width": 24,
        }
      }
    >
      <Connect(ProfilePicture)
        showStatus={false}
        size={24}
        testID="avatars.profile_picture"
        theme={
          Object {
            "awayIndicator": "#ffbc1f",
            "buttonBg": "#1c58d9",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3f4350",
            "codeTheme": "github",
            "dndIndicator": "#d24b4e",
            "errorTextColor": "#d24b4e",
            "linkColor": "#386fe5",
            "mentionBg": "#ffffff",
            "mentionColor": "#1e325c",
            "mentionHighlightBg": "#ffd470",
            "mentionHighlightLink": "#1b1d22",
            "newMessageSeparator": "#cc8f00",
            "onlineIndicator": "#3db887",
            "sidebarBg": "#1e325c",
            "sidebarHeaderBg": "#192a4d",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarTeamBarBg": "#14213e",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#5d89ea",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#28427b",
            "sidebarUnreadText": "#ffffff",
            "type": "Denim",
          }
        }
        userId="user1"
      />
    </View>
  </View>
</TouchableWithFeedbackIOS>
`;
